internal/pkg/model/component.go

```diff
@@ -61,6 +61,11 @@ func (c *ComponentsMap) Get(key ComponentKey) (*Component, error) {
 		return component, nil
 	}
 
+	// Remote provider can be nil in tests, prevent panic
+	if c.remoteProvider == nil {
```

Comment:
V unit testoch niekedy pre jednoduchost nastavime components `remoteProvider` na `nil`.
No niekedy je tato hodnota potrebna a skonci to s `panic` -> takze som to upravil, aby bola chybova hlaska zrejma.